src/Data.jsx

```diff
@@ -44,10 +49,19 @@ class Data extends React.Component {
   fetchData = overrides => {
     const { action, options, variables } = this.props;
     const { store } = this.context;
+    const fetchCount = this.state.fetchCount + 1;
+
+    // Creates a callback which only completes if fetchData hasn't been called
+    // again in the meantime
+    const makeCallback = (fn) => (...args) => {
+      if (this.state.fetchCount === fetchCount) fn(...args);
+    };
     const actionWithVariables = () => action(variables, this.context);
-    this.setState({ loading: true });
+
+    this.setState({ loading: true, fetchCount });
```

Comment:
i think you need to use functional set state here or youre going to hit race conditions